libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -333,6 +343,16 @@ public void onQueryCurrentProgressForUploadingMedia() {
                         updateMediaProgress();
                     }
                 },
+                new OnSetFeaturedImageListener() {
+                    @Override
+                    public void onSetFeaturedImageButtonClicked(int mediaId) {
+                        if (mediaId == 0 || mFeaturedImageId == 0) {
+                            setFeaturedImage(mediaId);
+                        } else if (mFeaturedImageId != mediaId) {
+                            showFeaturedImageConfirmationDialog(mediaId);
+                        }
+                    }
```

Comment:
Looks like the function only implicitly cover the case where the IDs are non-zero but equal. I know that that case is practically a no-op case (we don't want anything to happen if we're trying to set the same image as the current featured one) but it's best if we leave a comment for that to make it clear. Something like `// don't do anything if mediaId is the same as the current featuredImageId`.

Alternatively, and a bit of a nit, what do you think about restructuring the code in such a way that it's more clear what's going on with the special values (`0`) and the various combinations, like this:

```kotlin
public void onSetFeaturedImageButtonClicked(int mediaId) {
    if (mediaId == mFeaturedImageId) {
        // nothing special to do, trying to set the image that's already set as featured
        return;
    }

    if (mediaId == 0) {
        // user tries to clear the featured image setting
        setFeaturedImage(mediaId);
        return;
    }

    if (mFeaturedImageId == 0) {
        // current featured image is not set so, go ahead and set it to the provided one
        setFeaturedImage(mediaId);
        return;
    }

    // ask the user to confirm changing the featured image since there's already one set
    showFeaturedImageConfirmationDialog(mediaId);
}
```

Reply:
Implemented!

> It's of course more verbose, but that's actually part of the goal, to use verbosity for the sake of readability and simplifying understanding the combinations.

This is useful insight, thank you! I think knowing when to favour verbosity is something I'm going to need to work at.

Reply:
> I think knowing when to favour verbosity is something I'm going to need to work at.

Thanks for looking at this that way! And you are 100% correct with "when to favour" as it's always a balance or compromise between too verbose and too compacted without it being clear usually to which side one should lean to 😬